### PR TITLE
#5 create metamask connection

### LINKED ISSUE #5: Sign in to metamask from web app

### PR DESCRIPTION
**Pull request by ptrepkaVL**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,14 +1,22 @@
+import { Web3Provider } from '@ethersproject/providers';
+import { Web3ReactProvider } from '@web3-react/core';
 import React from 'react';
+import { createRoot } from 'react-dom/client';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
-import { createRoot } from 'react-dom/client';
+
+const getLibrary = (provider: any) => {
```
Comment (maciejczekaj) [inline src/index.tsx:8]: can we add types to the provider to get IntelliSense?

File: src/components/layout/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { useWeb3React } from '@web3-react/core';
+import { InjectedConnector } from '@web3-react/injected-connector';
+import React, { useEffect, useState } from 'react';
+
+const Header = () => {
+	const { activate, active, deactivate, library: provider } = useWeb3React();
+	const [userWalletAddress, setUserWalletAddress] = useState('');
+
+	useEffect(() => {
+		const getUserData = async () => {
```
Comment (maciejczekaj) [inline src/components/layout/Header.tsx]: there is no need to wrap this call in getUserData method. You could just do it directly in useEffect 😄

File: src/components/layout/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			if (active) {
+				const signer = await provider.getSigner();
+				const userWalletAddress = await signer.getAddress();
+				setUserWalletAddress(userWalletAddress);
+			} else {
+				setUserWalletAddress('');
+			}
+		};
+		getUserData();
+	}, [active]);
```
Comment (maciejczekaj) [inline src/components/layout/Header.tsx]: please add a dependency on the provider and signer as those are also objects that can be changed.
If you try to sign in with one wallet and then try to use another wallet you'll see an issue

File: src/components/layout/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+	const connectMetamask = async () => {
+		try {
+			const injected: InjectedConnector = new InjectedConnector({});
```
Comment (maciejczekaj) [inline src/components/layout/Header.tsx]: you don't have to put a type after a variable name when it can be determined from assignment operation.

**Review by maciejczekaj — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 1 above)
Comment (maciejczekaj) [reply src/index.tsx:8]: That's not necessary as we don't do anything with this provider except forwarding to external component

File: src/components/layout/Header.tsx (hunk 4 above)
Comment (ptrepkaVL) [reply src/components/layout/Header.tsx]: Oki thanks. Changed:)

File: src/components/layout/Header.tsx (hunk 2 above)
Comment (ptrepkaVL) [reply src/components/layout/Header.tsx]: After discussion stays like it is:)

File: src/components/layout/Header.tsx (hunk 3 above)
Comment (ptrepkaVL) [reply src/components/layout/Header.tsx]: Done

File: package.json [hunk 5]
```diff
@@ -3,60 +3,65 @@
...
+    "@testing-library/jest-dom": "^5.16.3",
+    "@testing-library/react": "^12.1.4",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.4.1",
+    "@types/node": "^16.11.26",
+    "@types/react": "^17.0.43",
+    "@types/react-dom": "^17.0.14",
+    "@web3-react/core": "^6.1.9",
+    "@web3-react/injected-connector": "^6.0.7",
+    "eslint-plugin-react-hooks": "^4.4.0",
```
Comment (maciejczekaj) [inline package.json]: przenieśmy to do devDependencies, bo nie jest potrzebne przy buildzie produkcyjnym :)

Comment (ptrepkaVL) [reply package.json]: Poprawione:)

File: src/components/layout/Header.tsx (hunk 2 above)
Comment (pobuchowskivl) [reply src/components/layout/Header.tsx]: a tak z ciekawosci, czemu definiujemy to w srodku useEffect? :D

**Review by pobuchowskivl — APPROVED** [review]

**Review by maciejczekaj — APPROVED** [review]

**Pull request merged**